bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -18,47 +18,624 @@ package v1alpha1
+	//+optional
+	KubeletPath string `json:"kubeletPath,omitempty"`
+
+	// KubeletArgs Customized flag for kubelet process
+	// +optional
+	Kubelet ComponentConfig `json:"kubelet,omitempty"`
+
+	// KubeProxyArgs Customized flag for kube-proxy process
+	// +optional
+	KubeProxy ComponentConfig `json:"kubeProxy,omitempty"`
+}
+
+// RKE2ConfigStatus defines the observed state of RKE2Config
+type RKE2ConfigStatus struct {
+	// Ready indicates the BootstrapData field is ready to be consumed
+	Ready bool `json:"ready,omitempty"`
+
+	BootstrapData []byte `json:"bootstrapData,omitempty"`
+
+	// DataSecretName is the name of the secret that stores the bootstrap data script.
+	// +optional
+	DataSecretName *string `json:"dataSecretName,omitempty"`
+
+	// FailureReason will be set on non-retryable errors
+	// +optional
+	FailureReason string `json:"failureReason,omitempty"`
+
+	// FailureMessage will be set on non-retryable errors
+	// +optional
+	FailureMessage string `json:"failureMessage,omitempty"`
+
+	// ObservedGeneration is the latest generation observed by the controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// Conditions defines current service state of the RKE2Config.
+	// +optional
+	Conditions clusterv1.Conditions `json:"conditions,omitempty"`
 }
 
-//+kubebuilder:object:root=true
-//+kubebuilder:subresource:status
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
 
-// Rke2Config is the Schema for the rke2configs API
-type Rke2Config struct {
+// RKE2Config is the Schema for the rke2configs API
+type RKE2Config struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   Rke2ConfigSpec   `json:"spec,omitempty"`
-	Status Rke2ConfigStatus `json:"status,omitempty"`
+	Spec   RKE2ConfigSpec   `json:"spec,omitempty"`
+	Status RKE2ConfigStatus `json:"status,omitempty"`
 }
 
-//+kubebuilder:object:root=true
+func (c *RKE2Config) GetConditions() clusterv1.Conditions {
+	return c.Status.Conditions
+}
+
+func (c *RKE2Config) SetConditions(conditions clusterv1.Conditions) {
+	c.Status.Conditions = conditions
+}
 
-// Rke2ConfigList contains a list of Rke2Config
-type Rke2ConfigList struct {
+// +kubebuilder:object:root=true
+
+// RKE2ConfigList contains a list of RKE2Config
+type RKE2ConfigList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
-	Items           []Rke2Config `json:"items"`
+	Items           []RKE2Config `json:"items"`
+}
+
+// DisabledItem selects a plugin Components to be disabled
+// +kubebuilder:validation:enum=rke2-coredns;rke2-ingress-nginx;rke2-metrics-server
+type DisabledItem string
+
+// CisProfile defines the CIS Benchmark profile to be activated in RKE2
+// +kubebuilder:validation:enum=cis-1.23
+type CisProfile string
+
+// Cni defines the Cni options for deploying RKE2
+// +kubebuilder:validation:enum=none;calico;canal;cilium
+type Cni string
```

Comment:
```suggestion
type CNI string
```

Reply:
```golang
const (
    Cillium CNI = "cillum"
)
```
we can create some constants to make it easier to manipulate the API programmatically.

Reply:
Yes, sure, however, the kubebuilder validation line should stay the same, right?

Reply:
yes, we are only adding constants to help people configure the provider from the code.